File: src/2025-02-22-00-51-50--a-deepseek-r1-test-220-10-1--2-1-huoshan/go/exercises/practice/beer-song/beer_song.go
package beer

import (
	"fmt"
	"strings"
)

func Song() string {
	verses, _ := Verses(99, 0)
	return verses
}

func Verses(start, stop int) (string, error) {
	if start < stop {
		return "", fmt.Errorf("start must be >= stop")
	}
	if start > 99 || stop < 0 {
		return "", fmt.Errorf("verse number out of range")
	}

	var verses []string
	for n := start; n >= stop; n-- {
		v, err := Verse(n)
		if err != nil {
			return "", err
		}
		verses = append(verses, v)
	}

	return strings.Join(verses, "\n\n") + "\n", nil
}

func Verse(n int) (string, error) {
	if n < 0 || n > 99 {
		return "", fmt.Errorf("invalid verse number %d", n)
	}

	var first, second string

	switch {
	case n == 0:
		first = "No more bottles of beer on the wall, no more bottles of beer."
		second = "Go to the store and buy some more, 99 bottles of beer on the wall."
	case n == 1:
		first = "1 bottle of beer on the wall, 1 bottle of beer."
		second = "Take it down and pass it around, no more bottles of beer on the wall."
	default:
		first = fmt.Sprintf("%d bottles of beer on the wall, %d bottles of beer.", n, n)
		
		remaining := n - 1
		bottles := "bottles"
		if remaining == 1 {
			bottles = "bottle"
		}
		action := "one"
		if n == 1 {
			action = "it"
		}
		
		if remaining > 0 {
			second = fmt.Sprintf("Take %s down and pass it around, %d %s of beer on the wall.", action, remaining, bottles)
		} else {
			second = "Take one down and pass it around, no more bottles of beer on the wall."
		}
	}

	return fmt.Sprintf("%s\n%s\n", first, second), nil
}

File: src/2025-03-16-08-42-59--a-deepclaude-full-python-0316-1/python/exercises/practice/beer-song/beer_song.py
def recite(start, take=1):
    """
    Recite the lyrics to "99 Bottles of Beer on the Wall".
    
    Args:
        start: The number of bottles to start with
        take: The number of verses to generate
    
    Returns:
        A list of strings representing the lyrics
    """
    lyrics = []
    
    end = max(1, start - take + 1)
    for bottle_num in range(start, end - 1, -1):
        # Only process positive bottle numbers in this loop
        if bottle_num <= 0:
            break
            
        # First line of verse
        if bottle_num == 1:
            lyrics.append(f"1 bottle of beer on the wall, 1 bottle of beer.")
        else:
            lyrics.append(f"{bottle_num} bottles of beer on the wall, {bottle_num} bottles of beer.")
        
        # Second line of verse
        next_bottle = bottle_num - 1
        if next_bottle == 0:
            lyrics.append("Take it down and pass it around, no more bottles of beer on the wall.")
        elif next_bottle == 1:
            lyrics.append(f"Take one down and pass it around, 1 bottle of beer on the wall.")
        else:
            lyrics.append(f"Take one down and pass it around, {next_bottle} bottles of beer on the wall.")
        
        # Add blank line between verses
        if bottle_num > start - take + 1:
            lyrics.append("")
    
    # Handle special case for "no more bottles"
    if start - take < 0 and start > 0:
        lyrics.append("")  # Add blank line before final verse only if we have previous verses
    
    if start - take + 1 <= 0:
        lyrics.append("No more bottles of beer on the wall, no more bottles of beer.")
        lyrics.append("Go to the store and buy some more, 99 bottles of beer on the wall.")
    
    return lyrics
